src/types.js

```diff
@@ -394,13 +549,18 @@ encode.MACSTRING = function(str, encoding) {
                 return undefined;
             }
         }
-
-        result.push(c);
+        result[i] = c;
```

Comment:
Do you know @fpirsch if ` result.push(c);` is faster than direct assignment `result[i] = c;`?
It looks like it's faster… and easier to read too :) http://stackoverflow.com/a/15650115/1345244